File: catkin_ws/src/bowser_lidar/nodes/puck_laser_listener.cpp
#include <ros/ros.h>
#include <sensor_msgs/LaserScan.h>
#include <string>
#include <iostream>

using namespace std;

void onScan(const sensor_msgs::LaserScan laser){
	int i=0;
	float onePoint = laser.ranges[i];
	float dangerZone = 2.0;
	if(onePoint >= laser.range_min && onePoint <= laser.range_max)
		if(onePoint < dangerZone)
			printf("Stay away plz.\n");
		else
			printf("\n");
}

int main(int argc, char** argv){
	ros::init(argc, argv, "laser_listener");
	ros::NodeHandle node;
	ros::Subscriber sub = node.subscribe("scan", 1000, &onScan);

	while(node.ok()){	
		ros::spin();
	}

	return 0;
}

File: igvc2019_lidar/nodes/puck_laser_listener.py
#!/usr/bin/env python
import rospy
from sensor_msgs.msg import LaserScan
from sensor_msgs.msg import Imu

class Lidar:
		def __init__(self, scan_topic="scan"):
			# Subscribe to the laser scan topic
			self.scan_sub = rospy.Subscriber(scan_topic, LaserScan, self.on_scan)

		def on_scan(self, laser):
			ranges = laser.ranges
			print len(ranges)
			for range in ranges:
				if not (range < laser.range_min or range > laser.range_max or range == 'inf'):
					if(range < 2.0):
						self.stop_it()
					else:
						print ("%f %f %f" % (laser.range_min, range, laser.range_max))

		def stop_it(self):
			print "You are way too close"
			rospy.spin()

def on_imu(imu):
	print "imu is good"

if __name__ == '__main__':
	rospy.init_node("laser_tf", anonymous = False)
	laser = Lidar()
	imu = rospy.Subscriber("/imu/imu", Imu, on_imu)
	rospy.spin()